### PR TITLE
Add statuscode error handling for rest service

### PR DESCRIPTION
**Pull request by gabbaxx**
Now error codes like 404 etc and socket errors like ECONNREFUSED are handled properly in the rest service.

### Status

* [ ] Under development
* [*] Waiting for code review
* [ ] Waiting for merge

### Information

* [ ] Contains breaking changes
* [ ] Contains new API(s)
* [ ] Contains documentation
* [*] Contains test

### CODE REVIEW

File: src/services/rest/http-client.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import url from 'url';
 // see getDefaultHTTPModule for additional imports due to webpack
 
 const IS_NODE = typeof window === 'undefined';
+const HTTP_200_OK_RANGE_START = 200;
```
Comment (peol) [inline src/services/rest/http-client.js]: Maybe a list of explicitly "accepted" status codes (e.g. `ACCEPTED_HTTP_STATUSES`)? They're not that many: https://httpstatuses.com/ - As an example, is `204` really acceptable here (don't we expect the response to contain something?)

File: src/services/rest/http-client.js [hunk 2]
```diff
@@ -39,20 +41,56 @@ export function responseHandler(opts, res, handleLog) {
...
-      // do not set out.obj
-    }
-    if (handleLog) {
-      handleLog({ msg: 'Received', data: { body: res.data } });
+
+    // Add additional fields compatible with swagger-client
+    res.status = res.statusCode;
+    res.statusText = res.statusMessage;
+
+    if (HTTP_200_OK_RANGE_START <= res.statusCode && res.statusCode <= HTTP_200_OK_RANGE_END) {
```
Comment (peol) [inline src/services/rest/http-client.js]: And `if (ACCEPTED_HTTP_STATUSES.indexOf(res.statusCode) > -1)`

File: src/services/rest/http-client.js (hunk 1 above)
Comment (gabbaxx) [inline src/services/rest/http-client.js]: I don't agree we should fail anything in the 200 range.
"2xx Success
This class of status codes indicates the action requested by the client was received, understood, accepted, and processed successfully"

File: src/services/rest/http-client.js [hunk 3]
```diff
@@ -39,20 +41,56 @@ export function responseHandler(opts, res, handleLog) {
...
+      } catch (ex) {
+        // do not set out.obj
+      }
+      if (handleLog) {
+        handleLog({ msg: 'Received', data: { body: res.data } });
+      }
+      opts.on.response(res);
+    } else {
+      if (handleLog) {
+        handleLog({ msg: 'Error', data: res });
```
Comment (marcusoffesson) [inline src/services/rest/http-client.js]: As this is a traffic log, should message say 'Received' instead?

File: src/services/rest/http-client.js (hunk 1 above)
Comment (peol) [inline src/services/rest/http-client.js]: Yes, but that doesn't mean _we_ (the consumers) can do anything with a 204 No Content response. Or?

**Pull request merged**